components/emoji_picker/emoji_picker_section.jsx

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2016-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import PureRenderMixin from 'react-addons-pure-render-mixin';
+import {FormattedMessage} from 'react-intl';
+
+export default class EmojiPickerSection extends React.Component {
+    static propTypes = {
+        categoryName: PropTypes.string.isRequired,
+        children: PropTypes.any,
+        updateCategoryOffset: PropTypes.func.isRequired
+    };
+
+    static defaultProps = {
+    };
+
+    constructor(props) {
+        super(props);
+
+        // All props are primitives or treated as immutable
+        this.shouldComponentUpdate = PureRenderMixin.shouldComponentUpdate.bind(this);
```

Comment:
Can we use `React.PureComponent` instead?

Reply:
done